src/components/inline_edit/inline_edit_form.test.tsx

```diff
@@ -0,0 +1,303 @@
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          isInvalid={true}
+        />
+      );
+
+      expect(container.firstChild).toMatchSnapshot();
+
+      expect(
+        getByTestSubject('euiInlineEditModeInput').hasAttribute('aria-invalid')
+      ).toBeTruthy();
+
+      expect(getByTestSubject('euiInlineEditModeSaveButton')).toBeDisabled();
+      expect(getByTestSubject('euiInlineEditModeCancelButton')).toBeTruthy();
+    });
+
+    it('returns the latest value within EuiFieldText upon saving', () => {
+      const onSaveFunction = jest.fn();
+
+      const { getByTestSubject } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+          onSave={onSaveFunction}
+        />
+      );
+
+      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+        target: { value: 'New message!' },
+      });
+      fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));
+
+      expect(onSaveFunction).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Toggling between readMode and editMode', () => {
+    it('clicking on the readModeButton takes us to editMode', () => {
+      const { getByTestSubject, queryByTestSubject } = render(
+        <EuiInlineEditForm {...commonInlineEditFormProps} />
+      );
+
+      fireEvent.click(getByTestSubject('euiInlineReadModeButton'));
+      expect(getByTestSubject('euiInlineEditModeInput')).toBeTruthy();
+      expect(queryByTestSubject('euiInlineReadModeButton')).toBeFalsy();
+    });
+
+    it('saves text and returns to readMode', () => {
+      const { getByTestSubject, getByText } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+        target: { value: 'New message!' },
+      });
+      expect(
+        getByTestSubject('euiInlineEditModeInput').getAttribute('value')
+      ).toEqual('New message!');
+      fireEvent.click(getByTestSubject('euiInlineEditModeSaveButton'));
+
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+      expect(getByText('New message!')).toBeTruthy();
+    });
+
+    it('cancels text and returns to readMode', () => {
+      const { getByTestSubject, getByText } = render(
+        <EuiInlineEditForm
+          {...commonInlineEditFormProps}
+          startWithEditOpen={true}
+        />
+      );
+
+      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
+        target: { value: 'New message!' },
+      });
+      expect(
+        getByTestSubject('euiInlineEditModeInput').getAttribute('value')
+      ).toEqual('New message!');
+      fireEvent.click(getByTestSubject('euiInlineEditModeCancelButton'));
+
+      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
+      expect(getByText('Hello World!')).toBeTruthy();
+    });
```

Comment:
Now that we've added the `onSave` prop, I'd also like to test that canceling does **not** fire onSave
```suggestion
    it('cancels text and returns to readMode', () => {
      const onSave = jest.fn();

      const { getByTestSubject, getByText } = render(
        <EuiInlineEditForm
          {...commonInlineEditFormProps}
          startWithEditOpen={true}
          onSave={onSave}
        />
      );

      fireEvent.change(getByTestSubject('euiInlineEditModeInput'), {
        target: { value: 'New message!' },
      });
      expect(
        getByTestSubject('euiInlineEditModeInput').getAttribute('value')
      ).toEqual('New message!');
      fireEvent.click(getByTestSubject('euiInlineEditModeCancelButton'));

      expect(getByTestSubject('euiInlineReadModeButton')).toBeTruthy();
      expect(getByText('Hello World!')).toBeTruthy();
    });
    expect(onSave).not.toHaveBeenCalled();
```